src/coins.h

```diff
@@ -127,6 +127,9 @@ struct CCoinsCacheEntry
         FRESH = (1 << 1),
     };
 
+    inline bool IsDirty() const { return flags & DIRTY; }
+    inline bool IsFresh() const { return flags & FRESH; }
```

Comment:
nit:
untested, but modern compilers will likely [decide the need for inlining themselves
](https://en.cppreference.com/w/cpp/language/inline#:~:text=Since%20inline%20substitution)
> Since inline substitution is unobservable in the standard semantics, compilers are free to use inline substitution for any function that's not marked inline, and are free to generate function calls to any function marked inline.

Reply:
Does it make sense to check what the main compilers do in this case (the ones I checked via godbolt automatically inline), or was that just a waste of time, since seemingly unrelated changes could change the outcome (but if that's the case, shouldn't we let the compiler decide)?

Reply:
@andrewtoth are you asking because of the hasher involved in caching, i.e. https://github.com/bitcoin/bitcoin/blob/master/src/util/hasher.h#L41-L50?

Reply:
@sipa thank you for the explanation.
@paplorinc that was an example where it made a difference to performance that came to my mind, yes. I was just curious whether there would be any performance benefit of adding `noexcept` on these methods. If the asm generated with or without is the same then the answer is no. But the advice for adding them for human documentation makes sense.

Reply:
Added `noexcept` everywhere.